newsroom/users/users.py

```diff
@@ -191,11 +198,23 @@ def check_permissions(self, doc, updates=None):
             return
         if is_current_user_admin() or is_current_user_account_mgr():
             return
+
+        # Only check against metadata that has changed from the original
+        updated_fields = (
+            []
+            if updates is None
+            else [field for field in updates.keys() if updates[field] != doc.get(field) and field != "id"]
+        )
+
         if is_current_user_company_admin():
             manager = get_user()
             if doc.get("company") and doc["company"] == manager.get("company"):
-                if not updates or all([key in COMPANY_ADMIN_ALLOWED_UPDATES for key in updates.keys()]):
+                if not updated_fields or all([key in COMPANY_ADMIN_ALLOWED_UPDATES for key in updated_fields]):
+                    return
+        if request and request.url_rule and request.url_rule.rule:
+            if request.url_rule.rule in ["/reset_password/<token>", "/token/<token_type>"]:
+                return
+            elif request.url_rule.rule == "/users/<_id>":
```

Comment:
just wondering - would this also affect the mgmt api?

Reply:
@petrjasek This shouldn't be an issue, as we're overriding the function in `cp.mgmt_api.user.CPUsersService.check_permission`
```python
class CPUsersService(UsersService):
    ....
    def check_permissions(self, doc, updates=None):
        """Avoid testing if user has permissions."""
        pass
```